scripts/ffcv/create_ffcv_datasets.py

```diff
@@ -158,10 +170,28 @@ def _main():
     else:
         remote = os.path.join(args.remote, args.split)
         local = os.path.join(args.local, args.split)
-        cache_streaming(remote=remote, local=local, num_workers=args.num_workers)
-        ds = StreamingImageClassDataset(remote=remote, local=local, shuffle=False, batch_size=1)
 
-    write_ffcv_dataset(dataset=ds,
+        # Build a simple StreamingDataset that reads in the raw images, no processing
+        # Differet datasets are created + encoded differently, so that's why we need different decoders
+        if args.dataset == 'cifar':
+            def _decode_image(data):
+                arr = np.frombuffer(data, np.uint8)
+                arr = arr.reshape(32, 32, 3)
+                return Image.fromarray(arr)
+            def _decode_class(data):
+                return np.frombuffer(data, dtype=np.int64)[0]
+        elif args.dataset == 'imagenet':
+            def _decode_image(data):
+                return Image.open(BytesIO(data)).convert('RGB')
+            def _decode_class(data):
+                return np.frombuffer(data, dtype=np.int64)[0]
+        decoders = {'x': _decode_image, 'y': _decode_class}
+        dataset = PILImageClassStreamingDataset(remote=remote, local=local, decoders=decoders, shuffle=False)
```

Comment:
@knighton @dblalock just wanted to draw some eyes here for future UX

Some of this could potentially be swept away if `StreamingDataset` has an auto decoder, but still, I think there's some issue of modularity...

If a dataset author writes+uploads files, and then they write a dataset reader, they're probably going to want to add in all their training processing steps, and the object they build, like `StreamingImageNet1k` will have all that processing stuff.

If a second person comes along and wants to read the files in a custom way, right now they have to build their own reader class, like here.

If we want modularity, then the original dataset reader author has to either be proactive and provide args for doing so, or maybe they have to build multiple "pipes" like `ImageNet1kReaderPipe`, and `ImageNet1kAugmentationPipe`, `ImageNet1kTrainPipe`, etc. Where the writing burden should fall is not clear to me. 

The only thing I'm feeling more certain of is that enforcing encodings / sample types is really brittle. Even ImageNet / CIFAR10 do not store their data in the same format, despite both being image classification datasets! That's why we have different decoders here. And the sample type (PIL, class_id) format is a standard from `torchvision`, but it breaks almost immediately once you get out of image classification land. 

FWIW I don't think this problem is unique to us, it's the same issue motivating `torchdata` towards lots of little pipes... and I guess that philosophy would say that the second person should write their own custom reader. Food for thought...